lumigator/python/mzai/jobs/inference/inference.py

```diff
@@ -0,0 +1,161 @@
+                Path(storage_path[5:]) / config.name / "inference_results.json"
+            )
+        logger.info(f"Storing into {storage_path}...")
+        s3.put_file(local_path, storage_path)
+
+    # generate local temp file ANYWAY
+    # (we don't want to lose all eval data if there is an issue wth s3)
+    local_path = Path(
+        Path.home() / ".lumigator" / "results" / config.name / "inference_results.json"
+    )
+
+    try:
+        save_to_disk(local_path)
+
+        # copy to s3 and return path
+        if storage_path is not None and storage_path.startswith("s3://"):
+            save_to_s3(local_path, storage_path)
+            return storage_path
+
+        return local_path
+
+    except Exception as e:
+        logger.error(e)
+
+
+def run_inference(config: Box) -> Path:
+    # Do something with your config here, for example:
+    print("Received configuration:")
+    print(json.dumps(config, indent=4))
+
+    # Load dataset given its URI
+    dataset = load_from_disk(config.dataset.path)
+
+    # Limit dataset length if max_samples is specified
+    max_samples = config.evaluation.max_samples
+    if max_samples and max_samples > 0:
+        if max_samples > len(dataset):
+            logger.info(f"max_samples ({max_samples}) resized to dataset size ({len(dataset)})")
+            max_samples = len(dataset)
+        dataset = dataset.select(range(max_samples))
+
+    # Enable / disable tqdm
+    input_samples = dataset["examples"]
+    dataset_iterable = tqdm(input_samples) if config.evaluation.enable_tqdm else input_samples
+
+    # Choose which model client to use
+    if config.model.inference is not None:
+        # a model *inference service* is passed
+        base_url = config.model.inference.base_url
+        output_model_name = config.model.inference.engine
+        if "mistral" in base_url:
+            # run the mistral client
+            logger.info(f"Using Mistral client. Endpoint: {base_url}")
+            model_client = MistralModelClient(base_url, config.model)
+        else:
+            # run the openai client
+            logger.info(f"Using OAI client. Endpoint: {base_url}")
+            model_client = OpenAIModelClient(base_url, config.model)
+    # else:
+    #     # a *huggingface model name* is passed
+    #     model_name = hf_model_loader.resolve_asset_path(config["model"]["path"])
+    #     output_model_name = config["model"]["path"]
+    #     # depending on config, use the summarizer pipeline or directly call the model
+    #     # for inference
+    #     if config.evaluation.use_pipeline:
+    #         logger.info(f"Using summarization pipeline. Model: {model_name}")
+    #         model_client = SummarizationPipelineModelClient(model_name, config)
+    #     else:
+    #         logger.info(f"Using direct HF model invocation. Model: {model_name}")
+    #         model_client = HuggingFaceModelClient(model_name, config)
+
+    # run inference
+    output = {}
```

Comment:
suggestion: put `output` at top of method so it's clearer to read "inputs" and return